homeassistant/components/ping/config_flow.py

```diff
@@ -0,0 +1,71 @@
+"""Config flow for Ping (ICMP) integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+import logging
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.helpers import selector
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+)
+
+from .const import CONF_IMPORTED_BY, CONF_PING_COUNT, DEFAULT_PING_COUNT, DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+OPTIONS_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): str,
+        vol.Optional(
+            CONF_PING_COUNT, default=DEFAULT_PING_COUNT
+        ): selector.NumberSelector(
+            selector.NumberSelectorConfig(
+                min=1, max=100, mode=selector.NumberSelectorMode.BOX
+            )
+        ),
+    }
+)
+
+
+CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): selector.TextSelector(),
+    }
+).extend(OPTIONS_SCHEMA.schema)
+
+CONFIG_FLOW = {"user": SchemaFlowFormStep(CONFIG_SCHEMA)}
+
+OPTIONS_FLOW = {
+    "init": SchemaFlowFormStep(OPTIONS_SCHEMA),
+}
+
+
+class ConfigFlowHandler(SchemaConfigFlowHandler, domain=DOMAIN):
+    """Handle a config or options flow for Ping."""
+
+    config_flow = CONFIG_FLOW
+    options_flow = OPTIONS_FLOW
+
+    def async_config_entry_title(self, options: Mapping[str, Any]) -> str:
+        """Return config entry title."""
+        return cast(str, options[CONF_NAME])
+
+    async def async_step_import(self, import_info: Mapping[str, Any]) -> FlowResult:
+        """Import an entry."""
+
+        to_import = {
+            CONF_NAME: import_info[CONF_NAME],
+            CONF_HOST: import_info[CONF_HOST],
+            CONF_PING_COUNT: import_info[CONF_PING_COUNT],
+            CONF_IMPORTED_BY: import_info[CONF_IMPORTED_BY],
```

Comment:
the source will already be SOURCE_IMPORT so I think this line is redundant

Reply:
Oh, I see you are combining `device_tracker` and `binary_sensor`

I guess this makes sense

Reply:
Yes, I combine them and only want to activate the sensor that the user has configured in YAML.